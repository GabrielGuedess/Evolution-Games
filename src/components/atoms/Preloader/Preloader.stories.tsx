import { Story, Meta } from '@storybook/react/types-6-0';

import { Preloader } from './Preloader';

export default {
  title: 'Atoms/Preloader',
  component: Preloader,
} as Meta;

export const Default: Story = () => <Preloader />;
