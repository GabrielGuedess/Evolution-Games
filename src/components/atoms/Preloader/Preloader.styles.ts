import styled, { keyframes } from 'styled-components';

const worm1 = keyframes`
    from {
      animation-timing-function: ease-in-out;
      stroke-dashoffset: -87.96;
    }
    20% {
      animation-timing-function: ease-in;
      stroke-dashoffset: 0;
    }
    60% {
      stroke-dashoffset: -791.68;
      visibility: visible;
    }
    60.1%,
    to {
      stroke-dashoffset: -791.68;
      visibility: hidden;
    }
`;

const worm2 = keyframes`
    from, 60% {
      stroke-dashoffset: -87.96;
      visibility: hidden;
    }
    60.1% {
      animation-timing-function: cubic-bezier(0, 0, 0.5, 0.75);
      stroke-dashoffset: -87.96;
      visibility: visible;
    }
    77% {
      animation-timing-function: cubic-bezier(0.5, 0.25, 0.5, 0.88);
      stroke-dashoffset: -340;
      visibility: visible;
    }
    to {
      stroke-dashoffset: -669.92;
      visibility: visible;
    }
`;

export const Wrapper = styled.div`
  display: flex;
  justify-content: center;
`;

export const Loader = styled.svg`
  width: 5em;
  height: 10em;

  .ap__ring {
    stroke: hsla(223, 10%, 90%, 0.1);
    transition: stroke 0.3s;
  }

  .ap__worm1,
  .ap__worm2 {
    animation-duration: 3s;
    animation-iteration-count: infinite;
  }
  .ap__worm1 {
    animation-name: ${worm1};
  }
  .ap__worm2 {
    animation-name: ${worm2};
    visibility: hidden;
  }
`;
