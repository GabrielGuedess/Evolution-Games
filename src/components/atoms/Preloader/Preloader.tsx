import { tint, darken } from 'polished';

import theme from 'styles/theme';

import * as S from './Preloader.styles';

export const Preloader = () => (
  <S.Wrapper>
    <S.Loader
      aria-label="Loading"
      viewBox="0 0 128 256"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs>
        <linearGradient id="ap-grad1" x1="0" y1="0" x2="0" y2="1">
          <stop offset="0%" stopColor={theme.colors.primary} />
          <stop offset="100%" stopColor={tint(0.1, theme.colors.primary)} />
        </linearGradient>
        <linearGradient id="ap-grad2" x1="0" y1="0" x2="0" y2="1">
          <stop offset="0%" stopColor={tint(0.3, theme.colors.primary)} />
          <stop offset="50%" stopColor={darken(0.1, theme.colors.primary)} />
          <stop offset="100%" stopColor={tint(0.1, theme.colors.primary)} />
        </linearGradient>
      </defs>

      <circle
        className="ap__ring"
        r="56"
        cx="64"
        cy="192"
        fill="none"
        stroke="#ddd"
        strokeWidth="16"
        strokeLinecap="round"
      />
      <circle
        className="ap__worm1"
        r="56"
        cx="64"
        cy="192"
        fill="none"
        stroke="url(#ap-grad1)"
        strokeWidth="16"
        strokeLinecap="round"
        strokeDasharray="87.96 263.89"
      />
      <path
        className="ap__worm2"
        d="M120,192A56,56,0,0,1,8,192C8,161.07,16,8,64,8S120,161.07,120,192Z"
        fill="none"
        stroke="url(#ap-grad2)"
        strokeWidth="16"
        strokeLinecap="round"
        strokeDasharray="87.96 494"
      />
    </S.Loader>
  </S.Wrapper>
);
