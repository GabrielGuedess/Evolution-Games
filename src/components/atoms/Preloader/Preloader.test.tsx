import { render, screen } from '@testing-library/react';

import { Preloader } from './Preloader';

describe('<Preloader />', () => {
  it('should render the heading', () => {
    const { container } = render(<Preloader />);

    expect(screen.getByLabelText(/Loading/)).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
