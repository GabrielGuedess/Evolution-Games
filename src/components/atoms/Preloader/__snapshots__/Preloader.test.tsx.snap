// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<Preloader /> should render the heading 1'] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 5em;
  height: 10em;
}

.c1 .ap__ring {
  stroke: hsla(223,10%,90%,0.1);
  -webkit-transition: stroke 0.3s;
  transition: stroke 0.3s;
}

.c1 .ap__worm1,
.c1 .ap__worm2 {
  -webkit-animation-duration: 3s;
  animation-duration: 3s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
}

.c1 .ap__worm1 {
  -webkit-animation-name: cmHHGy;
  animation-name: cmHHGy;
}

.c1 .ap__worm2 {
  -webkit-animation-name: ilqOlV;
  animation-name: ilqOlV;
  visibility: hidden;
}

<div
  class="c0"
>
  <svg
    aria-label="Loading"
    class="c1"
    viewBox="0 0 128 256"
    xmlns="http://www.w3.org/2000/svg"
  >
    <defs>
      <lineargradient
        id="ap-grad1"
        x1="0"
        x2="0"
        y1="0"
        y2="1"
      >
        <stop
          offset="0%"
          stop-color="#855aff"
        />
        <stop
          offset="100%"
          stop-color="#916aff"
        />
      </lineargradient>
      <lineargradient
        id="ap-grad2"
        x1="0"
        x2="0"
        y1="0"
        y2="1"
      >
        <stop
          offset="0%"
          stop-color="#a98bff"
        />
        <stop
          offset="50%"
          stop-color="#5f27ff"
        />
        <stop
          offset="100%"
          stop-color="#916aff"
        />
      </lineargradient>
    </defs>
    <circle
      class="ap__ring"
      cx="64"
      cy="192"
      fill="none"
      r="56"
      stroke="#ddd"
      stroke-linecap="round"
      stroke-width="16"
    />
    <circle
      class="ap__worm1"
      cx="64"
      cy="192"
      fill="none"
      r="56"
      stroke="url(#ap-grad1)"
      stroke-dasharray="87.96 263.89"
      stroke-linecap="round"
      stroke-width="16"
    />
    <path
      class="ap__worm2"
      d="M120,192A56,56,0,0,1,8,192C8,161.07,16,8,64,8S120,161.07,120,192Z"
      fill="none"
      stroke="url(#ap-grad2)"
      stroke-dasharray="87.96 494"
      stroke-linecap="round"
      stroke-width="16"
    />
  </svg>
</div>
`;
